pandas/tseries/tests/test_base.py

```diff
@@ -802,6 +802,37 @@ def test_na_value(self):
         self.assertIs(pd.DatetimeIndex._na_value, pd.NaT)
         self.assertIs(pd.DatetimeIndex([])._na_value, pd.NaT)
 
+    def test_equals(self):
+        # GH 13107
+        for tz in [None, 'UTC', 'US/Eastern', 'Asia/Tokyo']:
+            idx = pd.DatetimeIndex(['2011-01-01', '2011-01-02', 'NaT'])
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.asobject))
+            self.assertTrue(idx.asobject.equals(idx))
+            self.assertTrue(idx.asobject.equals(idx.asobject))
+            self.assertFalse(idx.equals(list(idx)))
+            self.assertFalse(idx.equals(pd.Series(idx)))
+
+            idx2 = pd.DatetimeIndex(['2011-01-01', '2011-01-02', 'NaT'],
+                                    tz='US/Pacific')
+            self.assertFalse(idx.equals(idx2))
+            self.assertFalse(idx.equals(idx2.copy()))
+            self.assertFalse(idx.equals(idx2.asobject))
+            self.assertFalse(idx.asobject.equals(idx2))
+            self.assertFalse(idx.equals(list(idx2)))
+            self.assertFalse(idx.equals(pd.Series(idx2)))
+
+            # same internal, different tz
+            idx3 = pd.DatetimeIndex._simple_new(idx.asi8, tz='US/Pacific')
+            tm.assert_numpy_array_equal(idx.asi8, idx3.asi8)
+            self.assertFalse(idx.equals(idx3))
+            self.assertFalse(idx.equals(idx3.copy()))
+            self.assertFalse(idx.equals(idx3.asobject))
```

Comment:
do we have tests for same local values but different tz not being equal? (similar to what you have above but more focused on local times)